pkg/oci/oci.go

```diff
@@ -706,6 +706,14 @@ func verifyImage(ctx context.Context, image string, imgOpts *ImageOptions) error
 		return fmt.Errorf("getting image digest: %w", err)
 	}
 
+	return VerifyImage(ctx, imageRef, imageDigest, imgOpts)
+}
+
+// VerifyImage is used to verify the image corresponding to the reference and
+// digest with the given public key.
+// If the image was signed with the given public key, nothing is returned,
+// otherwise an error is returned.
+func VerifyImage(ctx context.Context, imageRef reference.Named, imageDigest string, imgOpts *ImageOptions) error {
```

Comment:
I find it very confusing to have two functions with almost the same name and such a different signatures. Perhaps the contents of `verifyImage` can be inlined in `ensureImage`.

Reply:
Just to be sure we're on the same page, I was suggesting something like this:

```diff
diff --git a/pkg/oci/oci.go b/pkg/oci/oci.go
index 9ae4d6f24..0d17cd58e 100644
--- a/pkg/oci/oci.go
+++ b/pkg/oci/oci.go
@@ -690,25 +690,6 @@ func checkPayloadImage(payloadBytes []byte, imageDigest string) error {
 	return nil
 }
 
-func verifyImage(ctx context.Context, image string, imgOpts *ImageOptions) error {
-	imageStore, err := GetLocalOciStore()
-	if err != nil {
-		return fmt.Errorf("getting local oci store: %w", err)
-	}
-
-	imageRef, err := NormalizeImageName(image)
-	if err != nil {
-		return fmt.Errorf("normalizing image name: %w", err)
-	}
-
-	imageDigest, err := getImageDigest(ctx, imageStore, imageRef.String())
-	if err != nil {
-		return fmt.Errorf("getting image digest: %w", err)
-	}
-
-	return VerifyImage(ctx, imageRef, imageDigest, imgOpts)
-}
-
 // VerifyImage is used to verify the image corresponding to the reference and
 // digest with the given public key.
 // If the image was signed with the given public key, nothing is returned,
@@ -827,12 +808,17 @@ func ensureImage(ctx context.Context, imageStore oras.Target, image string, imgO
 		return nil
 	}
 
-	err := verifyImage(ctx, image, imgOpts)
+	imageRef, err := NormalizeImageName(image)
+	if err != nil {
+		return fmt.Errorf("normalizing image name: %w", err)
+	}
+
+	imageDigest, err := getImageDigest(ctx, imageStore.(*oci.Store), imageRef.String())
 	if err != nil {
-		return fmt.Errorf("verifying image %q: %w", image, err)
+		return fmt.Errorf("getting image digest: %w", err)
 	}
 
-	return nil
+	return VerifyImage(ctx, imageRef, imageDigest, imgOpts)
 }
 
 // EnsureImage ensures the image is present in the local store
```

> I can rename the verifyImage() to verifyOCIImage() or verifyGadgetImage().
This way, we have the generic VerifyImage() and the specialized verifyBuilderImage() and verifyGadgetImage() which rely on the first one.
What do you think?

That can work as well, but TBH I don't see a lot of value on having `verifyImage`, it's used in a single place and just handle a couple of operations that can be done directly be the caller.